src/generate/handlebarsAmfHelpers.ts

```diff
@@ -45,20 +45,33 @@ export const getBaseUriFromDocument = (
  */
 export const isTypeDefinition = (
   domainElement: model.domain.DomainElement
-): boolean => {
+): domainElement is model.domain.NodeShape => {
```

Comment:
What is this weird return type?

Reply:
It's a [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates). Basically `boolean`, but with additional information for the type system.